pkg/libs/endpointaccessible/endpoint_accessible_controller.go

```diff
@@ -94,12 +94,18 @@ func (c *endpointAccessibleController) sync(ctx context.Context, syncCtx factory
 		return err
 	}
 
-	newEndpoints := sets.New[string](endpoints...)
+	newEndpoints := sets.New(endpoints...)
 	endpointsChanged := !c.lastEndpoints.Equal(newEndpoints)
 
-	client, tlsChanged, err := c.buildTLSClient()
-	if err != nil {
-		return err
+	tlsChanged := false
+	var tlsConfig *tls.Config
+	if c.getTLSConfigFn != nil {
```

Comment:
do we allow this to be nil?

Reply:
This was like this [before](https://github.com/openshift/cluster-authentication-operator/pull/630/files#diff-e4aa3820af0ba868ac8a7d5e669e3dbf6c86939bc2af348f497dae427f62f66fL164) -- I've removed the check now, it is now used as `endpointListFn()` is, without any nil check.